ADVANCED.md

```diff
@@ -1260,6 +1237,43 @@ need to have in common is an `additionalOptions` parameter.
 For error handling, a method must throw any error with a "message" field populated for any unrecoverable error. This
 includes things like 404s, 500s, etc.
 
+## Performance
+
+Note that in order to make sure that components are as fast as possible, all components within
+this library are "Pure Components". You can read more about the concept and potential pitfalls
+of Pure Components in the React
+[Optimizing Performance](https://reactjs.org/docs/optimizing-performance.html#avoid-reconciliation) guide.
+
+The thing to be most cautious of is not to
+[mutate state](https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data) that you will
+pass as props to any of these components.
+
+Example of what not to do:
+
+```jsx
+class SomeComponent extends React.Component {
+  changeSorting = () => {
+    const { options } = this.state.options;
+    // Mutating an existing object in state rather than creating a new /// one is bad ... since Sorting component is "Pure"
+    // it won't update after call `setState` here.
+    options.push("newOption");
+    this.setState({ options });
+  };
+
+  render() {
+    const { options } = this.state;
+    return <Sorting options={options} />;
+  }
+}
+```
+
+Instead, do:
+
+```jsx
+const { options } = this.state.options;
+this.setState({ options: [...options, "newOption"] });
```

Comment:
```suggestion
this.setState(prevState => ({ options: [...prevState.options, "newOption"] }));
```

Reply:
Using prevState argument instead of previously saved state ensures that we work with the current state.